### DISCUSSION TITLE
Best way to handle "AggregateError" when connection is unavailable

### DISCUSSION THREAD

**Discussion by joshes** [dynamoose/dynamoose#1705] [Q&A] [answered]
If a connection fails for a request, the error is not being caught in my try/catch block and killing the application with an Unhandled Exception.

What is the correct way to check for a connection _without_ this happening.

A minimally viable reproduction follows:

```typescript
import dynamoose from "dynamoose"
import { it } from "vitest"
import { Item } from "dynamoose/dist/Item"

it("should not fail", async () => {
  // Use dynamodb-local (but don't start it up!)
  dynamoose.aws.ddb.local()

  // From docs
  class Cat extends Item {
    id: number
    name: string
  }

  try {
    const CatModel = dynamoose.model<Cat>("Cat", { id: Number, name: String })
    await CatModel.create({ id: 1, name: "Azura" })
  } catch (e) {
    console.log(e)
  }
})
```

Running the above results in with the try/catch not being effecitve.

```sh
AggregateError: 
    at internalConnectMultiple (node:net:1118:18)
    at afterConnectMultiple (node:net:1685:7)
```

I would very much like an unavailable network connection to kill my server.

What's the way forward?

**Comment by joshes** [comment]
Seems when running `vite dev`, its just a node server. Resolved it by adding the following to my `hooks.server.ts`

```typescript
if (process.setUncaughtExceptionCaptureCallback) {
  process.setUncaughtExceptionCaptureCallback((err) => {
    console.log(err)
  })
}
```

**Reply by fishcharlie** [reply, marked as answer]
That's one way to do it. But in production you should probably disable create, update, and waitForActive in your table settings. This is documented in the table settings page in the documentation and in the FAQ.

**Reply by joshes** [reply]
None of that is relevant when connecting to a `dynamodb-local` instance and calling a `get` is it? Maybe I'm misunderstanding something in how the connection is managed?

**Reply by fishcharlie** [reply]
When you create a table in Dynamoose if those settings are set to true it will automatically fire off network requests (to AWS or to DynamoDB Local depending on what you have configured). The AWS or DynamoDB Local is kinda irrelevant in this case. If those settings are true it'll call the AWS SDK and make requests.

**Reply by joshes** [reply]
Indeed that does the trick. Thanks @fishcharlie ✅
